### PR TITLE
[Feature][sql] Support variable in sql job conf 

### LINKED ISSUE #1309: [Feature][sql] Support variable in sql job conf
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support variable in sql job conf.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by simon824**
## Purpose of this pull request

closed #1309
Support variable in sql job conf 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-core/seatunnel-core-sql/src/test/java/org/apache/seatunnel/core/sql/SqlVariableSubstitution.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.seatunnel.core.sql.job.JobInfo;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+
+public class SqlVariableSubstitution {
```
Comment (yx91490) [inline seatunnel-core/seatunnel-core-sql/src/test/java/org/apache/seatunnel/core/sql/SqlVariableSubstitution.java]: better to change name to [SqlVariableSubstitutionTest.java]

File: seatunnel-core/seatunnel-core-sql/src/test/resources/flink.sql.conf.template [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+) WITH (
+  'connector' = 'datagen',
+  'rows-per-second'='5',
+  'fields.f_type.min'='1',
+  'fields.f_type.max'='5',
+  'fields.f_uid.min'='1',
+  'fields.f_uid.max'='1000'
+);
+
+CREATE TABLE print_table (
```
Comment (yx91490) [inline seatunnel-core/seatunnel-core-sql/src/test/resources/flink.sql.conf.template:40]: can you also change 'print_table' to a a variable?

Comment (xleoken) [reply seatunnel-core/seatunnel-core-sql/src/test/resources/flink.sql.conf.template:40]: emm, seems we don't need do this.

File: seatunnel-core/seatunnel-core-sql/src/test/java/org/apache/seatunnel/core/sql/SqlVariableSubstitution.java (hunk 1 above)
Comment (xleoken) [reply seatunnel-core/seatunnel-core-sql/src/test/java/org/apache/seatunnel/core/sql/SqlVariableSubstitution.java]: > better to change name to [SqlVariableSubstitutionTest.java]

Make sense.

**Review by xleoken — COMMENTED** [review]
Overall LGTM, please update the patch.

**Review by xleoken — APPROVED** [review]
cc @yx91490 @simon824 @asdf2014

**Pull request merged**